frontend/public/components/nav/__tests__/perspective-nav.spec.tsx

```diff
@@ -0,0 +1,149 @@
+  usePinnedResources: jest.fn(),
+}));
+jest.mock('@console/plugin-sdk/src/api/useExtensions', () => ({ useExtensions: jest.fn() }));
+jest.mock('react-dnd', () => {
+  const reactDnd = require.requireActual('react-dnd');
+  return {
+    ...reactDnd,
+    useDrag: jest.fn(() => [{}, {}]),
+    useDrop: jest.fn(() => [{}, {}]),
+  };
+});
+
+describe('Perspective Nav', () => {
+  beforeEach(() => {
+    useActivePerspectiveMock.mockClear();
+  });
+
+  it('should render Admin nav for admin perspective', () => {
+    useExtensionsMock.mockReturnValue(mockPerspectiveExtensions);
+    useActivePerspectiveMock.mockReturnValue(['admin', () => {}]);
+    usePinnedResourcesMock.mockReturnValue([['hgh'], setPinnedResourcesMock, true]);
+    const wrapper = shallow(<PerspectiveNav />);
+    expect(wrapper.find(AdminNav).exists()).toBe(true);
+  });
+  it('should render draggable pinned items for dev perspective', () => {
+    useActivePerspectiveMock.mockReturnValue(['dev', () => {}]);
+    usePinnedResourcesMock.mockReturnValue([
+      [
+        'core~v1~ConfigMap',
+        'build.openshift.io~v1~BuildConfig',
+        'apiregistration.k8s.io~v1~APIService',
+        'config.openshift.io~v1~Authentication',
+      ],
+      jest.fn(),
+      true,
+    ]);
+    const wrapper = shallow(<PerspectiveNav />);
+    expect(wrapper.find('[data-test-id="dev-perspective-nav"]').exists()).toBeTruthy();
+  });
+
+  it('should render non-draggable pinned items for dev perspective if only one pinned resource is available', () => {
+    useActivePerspectiveMock.mockReturnValue(['dev', () => {}]);
+    usePinnedResourcesMock.mockReturnValue([['core~v1~ConfigMap'], jest.fn(), true]);
+    const wrapper = shallow(<PerspectiveNav />);
+    expect(
+      wrapper
+        .find('[data-test-id="dev-perspective-nav"]')
+        .shallow()
+        .find(DragAndDrop)
+        .shallow()
+        .find(NavGroup)
+        .shallow()
+        .find(PinnedResource)
+        .shallow()
+        .find('[data-test-id="pinned-resource-item"]')
+        .exists(),
+    ).toBeTruthy();
+  });
+
+  it('should render draggable pinned items for dev perspective more than one pinned resource is available', () => {
+    useActivePerspectiveMock.mockReturnValue(['dev', () => {}]);
+    usePinnedResourcesMock.mockReturnValue([
+      [
+        'core~v1~ConfigMap',
+        'build.openshift.io~v1~BuildConfig',
+        'apiregistration.k8s.io~v1~APIService',
+      ],
+      jest.fn(),
+      true,
+    ]);
+    const wrapper = shallow(<PerspectiveNav />);
+    expect(
+      wrapper
+        .find('[data-test-id="dev-perspective-nav"]')
+        .shallow()
+        .find(DragAndDrop)
+        .shallow()
+        .find(NavGroup)
+        .shallow()
+        .find(PinnedResource)
+        .at(0)
+        .shallow()
+        .find('[data-test-id="draggable-pinned-resource-item"]')
+        .exists(),
+    ).toBeTruthy();
+  });
+
+  it('should handle invalid resources', () => {
```

Comment:
confused with what is this spec trying to test, can be more descriptive

Reply:
changed it to `should handle resource with invalid model`